### PR TITLE
[system-probe] Periodically update udp expiry timeout from sysctl

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

This PR adds the `pkg/network/config/sysctl` package, which reads and caches a sysctl value. This package is used in the network tracer for the udp connection expiry timeout. Currently the udp expiry timeout is read once on startup of the tracer. This change will allow the value to be updated from procfs once every minute.

### Motivation

The udp conntrack sysctl values are configurable, so we should update them regularly.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: pkg/network/config/sysctl/sysctl.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	v, updated, err := i.sctl.Get()
+	if err == nil && updated {
+		i.v, err = strconv.Atoi(v)
+	}
+
+	return i.v, err
+}
+
+type sctl struct {
+	ttl      int64
```
Comment (brycekahle) [inline pkg/network/config/sysctl/sysctl.go:66]: Any reason not to leave this a `time.Duration`? I think using the real type is much clearer.

File: pkg/network/config/sysctl/sysctl.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	lastRead int64
```
Comment (brycekahle) [inline pkg/network/config/sysctl/sysctl.go:67]: Using `time.Time` for this would also be preferable.

File: pkg/network/config/sysctl/sysctl.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+func newSCtl(procRoot, sysctl string, cacheFor time.Duration) *sctl {
+	return &sctl{
+		ttl:  int64(cacheFor.Seconds()),
+		path: filepath.Join(procRoot, "sys", sysctl),
+	}
+}
+
+func (s *sctl) Get() (string, bool, error) {
+	now := time.Now().Unix()
+	if s.lastRead > 0 && s.lastRead+s.ttl > now {
```
Comment (brycekahle) [inline pkg/network/config/sysctl/sysctl.go:80]: Using the full types, this could be `!s.lastRead.IsZero() && s.lastRead.Add(s.ttl).After(now)`

Comment (hmahmood) [reply pkg/network/config/sysctl/sysctl.go:80]: I actually thought this was a little too wordy for such a simple comparison. But no other particular reason for using int's.

Comment (brycekahle) [reply pkg/network/config/sysctl/sysctl.go:80]: `time.Duration` is actually a type alias of `int64`. In general, I prefer clear intent over conciseness.

File: pkg/network/config/sysctl/sysctl_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	sctl = NewString(procRoot, "foo", 2*time.Second)
+	v, err = sctl.Get()
+	require.NoError(t, err)
+	require.Equal(t, "bar", v)
+
+	createTmpSysctl(t, procRoot, "foo", "baz")
+	require.Eventually(t, func() bool {
+		v, _ := sctl.Get()
+		return v == "baz"
+	}, 3*time.Second, 500*time.Microsecond)
```
Comment (leeavital) [inline pkg/network/config/sysctl/sysctl_test.go:50]: For this kind of test, I prefer to to create a helper which takes in time.Now() and test the helper. 

```
func Get() foo {
 return getWithTime(time.Now())
}

func getWithTime(now time.Time) foo
}
```

This makes the test run faster. Only a 3 second speedup for this test, but slow tests add up.  It also makes the test less sensitive to the whims to the go scheduler.

**Review by brycekahle — APPROVED** [review]
